python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,498 @@
+# This file is part of drp_tasks.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ReprocessVisitImageTask", "ReprocessVisitImageConfig", "combine_backgrounds"]
+
+import lsst.afw.image as afwImage
+import lsst.afw.table as afwTable
+import lsst.meas.algorithms
+import lsst.meas.deblender
+import lsst.meas.extensions.photometryKron
+import lsst.meas.extensions.shapeHSM
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.pipe.base import connectionTypes
+from lsst.pipe.tasks import snapCombine
+
+
+class ReprocessVisitImageConnections(
+    pipeBase.PipelineTaskConnections, dimensions=("instrument", "visit", "detector")
+):
+    exposures = connectionTypes.Input(
+        doc="Exposure (or two snaps) to be calibrated, and detected and measured on.",
+        name="postISRCCD",
+        storageClass="Exposure",
+        multiple=True,  # to handle 1 exposure or 2 snaps
+        dimensions=["instrument", "exposure", "detector"],
+    )
+    visit_summary = connectionTypes.Input(
+        doc="Visit-level catalog summarizing all image characterizations and calibrations.",
+        name="finalVisitSummary",
+        storageClass="ExposureCatalog",
+        dimensions=["instrument", "visit"],
+    )
+    background_1 = connectionTypes.Input(
+        doc="Background models estimated during calibration.",
+        name="initial_pvi_background",
+        storageClass="Background",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    background_2 = connectionTypes.Input(
+        doc="Background that was fit on top of background_1.",
+        name="skyCorr",
+        dimensions=("instrument", "visit", "detector"),
+        storageClass="Background",
+    )
+
+    # TODO: we'll want the finalized catalog to get calib_psf_* from.
+    # TODO: eventually we'll want to pull in the STREAK mask from the diffim.
+
+    # outputs
+    source_schema = connectionTypes.InitOutput(
+        doc="Schema of the output source catalog.",
+        name="source_schema",
+        storageClass="SourceCatalog",
+    )
```

Comment:
Is this needed?  Does this task get chained with anything further downstream?

Reply:
I assume that all afw output catalogs need a schema `InitOutput`, since you have to define the schema in the Task `init` anyway.

Reply:
I guess it doesn't hurt, but my point is that this is used when we're chaining these things together and this seems like a "terminus" kind of task.

Reply:
You'd have thought that for `finalizeCharacterize`, too, but in this task I want to use its schema, to grab the definitions of the `calib_*` fields! So, I figured it was easier to just add it now.